File: macos/Scanner/Sources/Scanner/Detectors/GeminiCliDetector.swift
public struct GeminiCliDetector: Detector {
    public let id = "gemini-cli"
    public let displayName = "Gemini CLI"

    public init() {}

    public func collectFacts(fs: any FilesystemFacade) throws -> ScanFacts {
        let settingsPath = homePath(fs, ".gemini/settings.json")
        let credentialsPath = homePath(fs, ".gemini/credentials")
        let oauthPath = homePath(fs, ".gemini/oauth_credentials.json")
        var facts = ScanFacts()
        appendAppInstallationFact(
            &facts,
            fs: fs,
            appId: id,
            candidatePaths: [homePath(fs, ".gemini")]
        )

        if let text = try readIfPresent(fs, path: settingsPath) {
            facts.append(try FactParsers.parseGeminiSettings(text, appId: id, configPath: settingsPath))
            appendConfigFileFact(&facts, fs: fs, appId: id, path: settingsPath)
        }

        if fs.fileExists(credentialsPath) {
            facts.authFiles.append(AuthFileFact(appId: id, filePath: credentialsPath))
        }
        if fs.fileExists(oauthPath) {
            facts.authFiles.append(AuthFileFact(appId: id, filePath: oauthPath))
        }

        return facts
    }
}

File: windows/Scanner/Detectors/GeminiCliDetector.cs
using AIExposureScanner.Scanner.Filesystem;
using AIExposureScanner.Scanner.Models;
using AIExposureScanner.Scanner.Parsing;

namespace AIExposureScanner.Scanner.Detectors;

public sealed class GeminiCliDetector : IDetector
{
    public string Id => "gemini-cli";
    public string DisplayName => "Gemini CLI";

    public ScanFacts CollectFacts(IFilesystemFacade fs)
    {
        var settingsPath = DetectorSupport.HomePath(fs, ".gemini/settings.json");
        var credentialsPath = DetectorSupport.HomePath(fs, ".gemini/credentials");
        var oauthPath = DetectorSupport.HomePath(fs, ".gemini/oauth_credentials.json");
        var facts = new ScanFacts();
        DetectorSupport.AppendAppInstallationFact(
            facts,
            fs,
            Id,
            [DetectorSupport.HomePath(fs, ".gemini")]
        );

        var text = DetectorSupport.ReadIfPresent(fs, settingsPath);
        if (text is not null)
        {
            facts.Append(FactParsers.ParseGeminiSettings(text, Id, settingsPath));
            DetectorSupport.AppendConfigFileFact(facts, fs, Id, settingsPath);
        }

        if (fs.FileExists(credentialsPath))
        {
            facts.AuthFiles.Add(new AuthFileFact(Id, credentialsPath));
        }
        if (fs.FileExists(oauthPath))
        {
            facts.AuthFiles.Add(new AuthFileFact(Id, oauthPath));
        }

        return facts;
    }
}
